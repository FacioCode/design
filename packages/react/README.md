# Facio Design System components for React

[![lerna](https://img.shields.io/badge/maintained%20with-lerna-cc00ff.svg)](https://lerna.js.org/)
[![npm package](https://img.shields.io/npm/v/@faciocode/react/latest.svg)](https://www.npmjs.com/package/@faciocode/react)

## 

### Getting started

Install and get started with Facio Design System components for React for web applications.

#### 1. How to install

If your project uses `react` and `react-dom` version `16.14.0` or higher, run:

```shell script
npm i --save @faciocode/react
```

The [`@faciocode/react` npm package](https://www.npmjs.com/package/@faciocode/react) installs all Facio Design System UI components for React that you need.

This command will use the latest stable release. Please, always use `latest` distribution channel.

##### Not a React-based project?

Install our style tokens separately:

```shell script
npm i --save @faciocode/styles
```

#### 2. How to load the fonts

This package does not provide "Open Sans" font, only its font family names.

Load the Open Sans font with `400`, `600`, `700` and `800` font weights:

```html
<link href="https://fonts.gstatic.com"
      rel="preconnect" />
<link href="https://fonts.googleapis.com/css2?family=Open+Sans:wght@400;600;700;800&display=swap"
      rel="stylesheet">
```

#### 3. How to use

A quick example to get you started:

(Note that we enclose all your application in a `ThemeProvider` component)

```jsx
import {
  CssBaseline,
  ThemeProvider
} from "@faciocode/react";

export const App = () => {
  <ThemeProvider>
    <CssBaseline />
    <ContainedButton color={"brand"}>
      Hello Button
    </ContainedButton>
  </ThemeProvider>
} 
```

Check our [Design System Storybook](http://storybook.facio.dev/) for all React components.

This package only works fine with the use of `<ThemeProvider />`, **essential** for applying styles correctly.

Please do not try to override Facio Design System theme palette or another style tokens. Use the Design System themes and components as-is.

### Additional information

- Please note the package peer dependencies `react >= 16.14.0` and `react-dom >= 16.14.0`;
- We recommend at least LTS Node version (`v14.15.4+`);
- If your project uses TypeScript, we need TypeScript `4.1.2` at least;

#### Bundled in this package

Installing `@faciocode/react` will also install the following packages:

- `@faciocode/mui-styles`
- `@faciocode/styles`
- `@material-ui/core`
- Other dependencies for styling.

You don't need to install them separately to use the Design System.

### Questions?

For how-to questions and other issues, please use the [issues section on our GitHub repo](https://github.com/faciocode/design/issues).

### Changelog

Just upgraded? Please read the [Changelog](https://github.com/FacioCode/design/blob/main/packages/react/CHANGELOG.md).
