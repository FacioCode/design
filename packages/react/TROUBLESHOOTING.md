# Troubleshooting

## Fixing build errors

**Given** you have gatsby website with `@faciocode/react` package installed  
**When** you run `npm start` and try to open website, and you find the following message:

```
Failed to compile
./node_modules/@faciocode/react/dist/index.umd.js
Module not found: Error: Can't resolve '@material-ui/core/Box' in 'node_modules/@faciocode/react/dist'
This error occurred during the build time and cannot be dismissed.
```

**Then** check if you have installed the latest version of `@faciocode/react`

```shell
npm install @faciocode/react@latest
```
