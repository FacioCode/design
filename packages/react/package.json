{
	"name": "@faciocode/react",
	"displayName": "FacioReact",
	"version": "5.1.2",
	"private": false,
	"description": "React components for Facio",
	"keywords": [
		"react",
		"components",
		"facio",
		"design",
		"ds",
		"components"
	],
	"author": {
		"email": "technology@facio.com",
		"name": "Facio",
		"url": "https://github.com/FacioCode"
	},
	"homepage": "https://github.com/FacioCode/design/tree/main/packages/react#readme",
	"license": "ISC",
	"directories": {
		"lib": "dist",
		"source": "src"
	},
	"files": [
		"dist",
		"README.md",
		"dist/index.d.ts"
	],
	"esmodule": "dist/es/index.js",
	"jsdelivr": "dist/umd/index.js",
	"jsnext:main": "dist/es/index.js",
	"main": "dist/index.js",
	"module": "dist/es/index.js",
	"source": "src/index.ts",
	"types": "dist/index.d.ts",
	"typings": "dist/index.d.ts",
	"umd:main": "dist/umd/index.js",
	"unpkg": "dist/umd/index.js",
	"repository": {
		"directory": "packages/react",
		"type": "git",
		"url": "https://github.com/FacioCode/design"
	},
	"bugs": {
		"url": "https://github.com/FacioCode/design/issues"
	},
	"peerDependencies": {
		"@types/react": "^17.0.0",
		"react": "^16.14.0 || ^17.0.1",
		"react-dom": "^16.14.0 || ^17.0.1"
	},
	"dependencies": {
		"@faciocode/mui-themes": "^5.1.2",
		"@faciocode/styles": "^5.1.2",
		"@material-ui/core": "deprecations",
		"@material-ui/lab": "4.0.0-alpha.57",
		"clsx": "1.1.1",
		"lodash": "4.17.20"
	},
	"scripts": {
		"build": "npm run clean && npm run compile",
		"build-storybook": "build-storybook -c ./.storybook --quiet -o ./dist-storybook",
		"chromatic": "chromatic --output-dir dist-storybook",
		"chromatic-ci": "npm run chromatic --exit-once-uploaded",
		"clean": "rm -rf dist/",
		"compile": "npm run rollup && npm run emitDeclaration",
		"emitDeclaration": "tsc --declaration --declarationDir dist --emitDeclarationOnly",
		"rollup": "rollup -c",
		"start-storybook": "start-storybook --no-manager-cache -c ./.storybook -p 9009",
		"test": "echo \"Todo: write tests\" && exit 0"
	},
	"publishConfig": {
		"access": "public"
	}
}
