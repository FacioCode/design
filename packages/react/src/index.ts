export {
  Alert,
  AlertProps,
  AlertTitle,
  AlertTitleProps,
} from "./Alert";
export { AppBar, AppBarProps } from "./AppBar";
export { default as BottomSheet, BottomSheetProps } from "./BottomSheet";
export { ButtonGroup, ButtonGroupProps } from "./Button";
export {
  ContainedButton,
  ContainedButtonProps,
  CloseButton,
  CloseButtonProps,
  LinkButton,
  LinkButtonProps,
  OutlinedButton,
  OutlinedButtonProps,
} from "./Button";
export { Card, CardContent, CardProps } from "./Card";
export { Checkbox, CheckboxProps } from "./Checkbox";
export { Chip, ChipProps } from "./Chip";
export { default as Container, ContainerProps } from "./Container";
export { CssBaseline } from "./CssBaseline";
export { Currency, CurrencyProps } from "./Currency";
export {
  Dialog,
  DialogActions,
  DialogActionsProps,
  DialogContent,
  DialogContentProps,
  DialogContentText,
  DialogContentTextProps,
  DialogProps,
  DialogTitle,
  DialogTitleProps,
} from "./Dialog";
export { Divider, DividerProps } from "./Divider";
export { default as FormLabel, FormLabelProps } from "./FormLabel";
export { default as FormHelperText, FormHelperTextProps } from "./FormHelperText";
export { IconButton, IconButtonProps } from "./IconButton";
export { Input, InputProps } from "./Input";
export { default as InputBase, InputBaseProps } from "./InputBase";
export { default as InputLabel, InputLabelProps } from "./InputLabel";
export { default as LinearProgress, LinearProgressProps } from "./LinearProgress";
export { Link, LinkProps } from "./Link";
export { default as Paper, PaperProps } from "./Paper";
export { default as Popover, PopoverProps } from "./Popover";
export { default as Skeleton, SkeletonProps } from "./Skeleton";
export { Snackbar, SnackbarProps } from "./Snackbar";
export { default as Switch, SwitchProps } from "./Switch";
export {
  Table,
  TableBody,
  TableBodyProps,
  TableCell,
  TableCellProps,
  TableContainer,
  TableContainerProps,
  TableFooter,
  TableFooterProps,
  TableHead,
  TableHeadProps,
  TableProps,
  TableRow,
  TableRowProps,
} from "./Table";
export {
  default as Tabs,
  Tab,
  TabProps,
  TabPanel,
  TabPanelProps,
  TabsProps,
} from "./Tabs";
export { default as Toolbar, ToolbarProps } from "./Toolbar";
export { TextField, TextFieldProps } from "./TextField";
export { ThemeProvider, ThemeProviderProps } from "./ThemeProvider";
export { default as Time, TimeProps } from "./Time";
export { default as Tooltip, TooltipProps } from "./Tooltip";
export { Typography, TypographyProps } from "./Typography";

export { default as useMediaQuery } from "./hooks/useMediaQuery";
export { default as useTheme } from "./hooks/useTheme";

export { default as ServerStyleSheets } from "./ServerStyleSheets";

export { default as CloseIcon } from "./icons/CloseIcon";
export { default as WhatsAppIcon } from "./icons/WhatsAppIcon";
