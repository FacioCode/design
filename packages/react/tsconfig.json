{
  "compilerOptions": {
    "types": ["@emotion/core"]
  },
  "extends": "../../tsconfig"
}
