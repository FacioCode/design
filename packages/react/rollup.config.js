import { unpkg as browser, directories, displayName, module, source } from "./package.json";
import config from "../../config/rollup/rollup.config";

export default config({
  browser,
  directories,
  displayName,
  globals: {
    "@faciocode/styles": "FacioStyles",
    "@material-ui/core": "MuiCore",
    "@material-ui/core/styles": "MuiCoreStyles",
    "@material-ui/lab": "MuiLab",
    lodash: "_",
    "prop-types": "PropTypes",
    react: "React",
  },
  module,
  source,
});
