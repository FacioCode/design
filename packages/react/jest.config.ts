import { baseConfig } from "../../jest.config";
import { compilerOptions } from "./tsconfig.json";
import { pathsToModuleNameMapper } from "ts-jest/utils";
import pkg from "./package.json";

const config = {
  ...baseConfig,
  displayName: pkg.displayName,
  moduleNameMapper: pathsToModuleNameMapper(compilerOptions.paths, {
    prefix: "<rootDir>/",
  }),
};

export default config;
