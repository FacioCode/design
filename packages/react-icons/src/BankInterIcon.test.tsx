/* eslint-disable sort-imports */
import * as React from "react";
import { BankInterIcon } from "./BankInterIcon";
import { create } from "react-test-renderer";

describe("<BankInterIcon component", () => {
  it("should match snapshot", () => {
    const testRenderer = create(<BankInterIcon />);

    expect(testRenderer).toMatchSnapshot();
  });
});
