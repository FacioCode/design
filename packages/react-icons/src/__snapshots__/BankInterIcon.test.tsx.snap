// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BankInterIcon component should match snapshot 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root"
  focusable="false"
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M7.376 10.044c-.893 0-1.444.537-1.72 1.065l-.064-.892H4.268v5.609h1.388v-3.152c0-.93.485-1.5 1.218-1.5.732 0 1.004.493 1.004 1.348v3.304h1.388v-3.609c.002-1.252-.532-2.173-1.89-2.173zm13.68 1.065c.353-.702 1.072-1.065 1.613-1.065.096 0 .248.012.331.032v1.098h-.021c-.987 0-1.54.604-1.54 1.63v3.044h-1.771v-5.63h1.345l.043.89zm-4.802-1.065c1.828 0 2.734 1.357 2.638 3.325H14.9c.072 1.01.538 1.522 1.363 1.522.697 0 1.067-.342 1.226-.826h1.357C18.582 15.252 17.659 16 16.254 16c-1.734 0-2.78-1.164-2.78-2.912v-.143c0-1.726 1.069-2.901 2.78-2.901zm-4.702-1.696v1.87h1.452v.956h-1.452v2.913c0 .506.192.735.798.735.254 0 .504-.072.654-.126v1.174c-.254.086-.643.13-.961.13-1.275 0-1.88-.65-1.88-1.774V8.348h1.389zm-8.224 1.87v5.608H1.94v-4.653H1v-.956h2.328zm12.915.956c-.76 0-1.215.415-1.343 1.24l2.598-.023c-.075-.769-.483-1.217-1.255-1.217zM2.691 8c.37.013.666.298.68.654v.214a.692.692 0 0 1-.68.653h-.114a.692.692 0 0 1-.68-.653v-.214A.692.692 0 0 1 2.578 8z"
    fillRule="evenodd"
  />
</svg>
`;
